gui

```diff
@@ -502,49 +458,111 @@ if [[ "$guimode" == yad* ]];then
+    (
+    if [ "$1" != 0 ];then
+      sleep 1.5
+    fi
+    pkill -P "$detailspid" 2>/dev/null
+    kill "$detailspid" 2>/dev/null
+    ) &
+  }
+  
+  #constantly read from the pipe and send all output to yad
+  #if an item is selected, the listening while loop will act on the input given to it
+  # Yad app list window sends back user selections to this listening while loop
+  while read -r input ;do
+    if [[ "$input" == *'Updates/' ]];then
+      #updater window
+      kill_details
+      
+      get_positions
+      "${DIRECTORY}/updater" gui fast $geometry2 --skip-taskbar --close-on-unfocus
+      exitcode=$? #0 if update successful, 1 if list of updates was closed
       
-    3) #search
-      app="$(yadflags+=("$geometry2") ; app_search_gui)"
+      #figure out which prefix we are on and refresh the app list
+      prefix="$(echo "$input" | sed 's+Updates/$++g')"
       
-      if [ ! -z "$app" ];then
-        "${DIRECTORY}/gui" '' "$app" "/$app" "$app"
+      if [ $exitcode == 0 ];then
+        refresh_list
       fi
-      ;;
-    *) #unknown button
-      exit 0
-      ;;
-    esac
-  done
-  #END of yad main window loop
+    elif [ "$input" == 'Search/' ];then
+      #search window
+      kill_details
+      
+      details_window_check(){
+        local input
+        read input
+        [ -z "$input" ] && return
+        details_window "/$input"
+      }
+      get_positions
+      #show app details window for this selected app
+      yadflags+=("$geometry2" --close-on-unfocus)
+      app_search_gui | details_window_check &
+      detailspid=$!
+      
+    elif [ "$input" == 'Settings/' ];then
+      break
+    elif [[ "$input" == */ ]];then
+      #app folder selected
+      echo -e '\f' > $pipe #clear the yad list
+      "${DIRECTORY}/preload" yad "${input::-1}" >> $pipe #send yad the app-list - derived from folder name but dropping the final slash character
+      
+    elif [[ "$input" == 'YAD_XID is'* ]];then
+      YAD_XID="$(echo "$input" | awk '{print $3}')"
+    elif [ ! -z "$input" ];then #app details window
+      
+      get_positions
+      
+      #show app details window, and close the previous one
+      kill_details
+      details_window "$input" &
+      detailspid=$!
+    fi
+  done < <(yad "${yadflags[@]}" \
+    --image="${DIRECTORY}/icons/logo-64.png" --image-on-top --text "$motd" \
+    --list --no-headers --column=:IMG  --column=Name --column=Sysname:HD --column=tip:HD --column=@fore@:HD \
+    --separator='\n' --window-icon="${DIRECTORY}/icons/logo.png" \
+    --tooltip-column=4 \
+    --dclick-action "bash -c "\""return_sysname %s"\""" --select-action="bash -c "\""return_sysname %s"\""" \
+    --button="!${DIRECTORY}/icons/search.png"!'Search':"bash -c "\""return_sysname '' '' Search/"\""" \
+    --button="!${DIRECTORY}/icons/options.png"!'Settings':"bash -c "\""echo Settings/ ; kill "\$"YAD_PID"\""" \
+    "$geometry1" < <(tail -f --retry $pipe))
```

Comment:
you removed this https://github.com/Botspot/pi-apps/commit/ac8ee8a6505ab2a7cdc85341bca4d21ca6416f74 workaround

I image it is still necessary and will have to check. Its purpose was to kill the tail command when the yad dialog closed.

Reply:
hmm... actually this method seems to avoid the issue entirely. interesting.

Reply:
added it back with https://github.com/Botspot/pi-apps/pull/2370/commits/8efbbe90534d9d0fe12f6c047374b41bfce10d89